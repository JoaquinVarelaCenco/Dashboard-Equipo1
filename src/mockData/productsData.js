module.exports = [
    {
      "id": 2,
      "title": "iPhone X",
      "description": "SIM-Free, Model A19211 6.5-inch Super Retina HD display with OLED technology A12 Bionic chip with ...",
      "price": 899,
      "rating": { "rate": 4.44, "count": 126 },
      "stock": 34,
      "category": "smartphones",
      "images": [],
      "lastModified": "2022-10-12T01:07:11.630Z"
    },
    {
      "id": 3,
      "title": "Samsung Universe 9",
      "description": "Samsung's new variant which goes beyond Galaxy to the Universe",
      "price": 1249,
      "rating": { "rate": 4.09, "count": 845 },
      "stock": 36,
      "category": "smartphones"
    },
    {
      "id": 23,
      "title": "Orange Essence Food Flavou",
      "description": "Specifications of Orange Essence Food Flavour For Cakes and Baking Food Item",
      "price": 14,
      "rating": { "rate": 4.85, "count": 1645 },
      "stock": 26,
      "category": "groceries",
      "images": [
        "https://dummyjson.com/image/i/products/23/1.jpg",
        "https://dummyjson.com/image/i/products/23/2.jpg",
        "https://dummyjson.com/image/i/products/23/3.jpg",
        "https://dummyjson.com/image/i/products/23/4.jpg",
        "https://dummyjson.com/image/i/products/23/thumbnail.jpg"
      ]
    },
    {
      "id": 24,
      "title": "cereals muesli fruit nuts",
      "description": "original fauji cereal muesli 250gm box pack original fauji cereals muesli fruit nuts flakes breakfast cereal break fast faujicereals cerels cerel foji fouji",
      "price": 46,
      "rating": { "rate": 4.94, "count": 11378 },
      "stock": 113,
      "category": "groceries",
      "images": [
        "https://dummyjson.com/image/i/products/24/1.jpg",
        "https://dummyjson.com/image/i/products/24/2.jpg",
        "https://dummyjson.com/image/i/products/24/3.jpg",
        "https://dummyjson.com/image/i/products/24/4.jpg",
        "https://dummyjson.com/image/i/products/24/thumbnail.jpg"
      ],
      "lastModified": "2022-10-17T14:31:36.884Z"
    },
    {
      "id": 25,
      "title": "Gulab Powder 50 Gram",
      "description": "Dry Rose Flower Powder Gulab Powder 50 Gram â€¢ Treats Wounds",
      "price": 70,
      "rating": { "rate": 4.87, "count": 568 },
      "stock": 47,
      "category": "groceries",
      "images": [
        "https://dummyjson.com/image/i/products/25/1.png",
        "https://dummyjson.com/image/i/products/25/2.jpg",
        "https://dummyjson.com/image/i/products/25/3.png",
        "https://dummyjson.com/image/i/products/25/4.jpg",
        "https://dummyjson.com/image/i/products/25/thumbnail.jpg"
      ]
    }
  ]
  