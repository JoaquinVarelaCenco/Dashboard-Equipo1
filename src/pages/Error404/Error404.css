.error404__Main{
    display: flex;
    flex-direction: column;
    text-align: center;
    height: calc(100vh - 72px);
    width: 100%;
}

.error404__top{
    display: flex;
    justify-content: center;
    align-items: flex-end;
    width: 100%;
    height: 50%;
    background-color: aliceblue;
}

.error404__top h2{
    margin-bottom: -75px;
    font-size: 350px;
    font-weight: bold;
    color:gainsboro;
}

.error404__bottom{
    display: flex;
    flex-direction: column;
    align-items: center;
    gap: 12px;
    padding: 50px;
    width: 100%;
    height: 50%;
    background-color: gainsboro;
}

.error404__bottom p{
    font-size: 35px;
}

.error404__bottom p:nth-child(2){
    font-size: 22px;
    text-transform: uppercase;
}

.error404__bottom button{
    background-color: #5b5d5d ;
    border: none;
    padding: 8px 10px;
    font-size: 16px;
    color: white;
    margin: 0 6px;
    border-radius: 15px;
    min-width: 150px;
}

@media (max-width: 690px){
    .error404__top h2{
        font-size: 200px;
        margin-bottom: -50px;
    }
    .error404__bottom p{
        font-size: 25px;
    }
    
    .error404__bottom p:nth-child(2){
        font-size: 16px;
        text-transform: uppercase;
    }
}

/* DarkMode */

.dark .error404__top{
    background-color: var(--dark-Background);
}

.dark .error404__bottom{
    background-color: var(--dark-BlackLight);
}

.dark .error404__top h2{
    color: var(--dark-BlackLight);
}