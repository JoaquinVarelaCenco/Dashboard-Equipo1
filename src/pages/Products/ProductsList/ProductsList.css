.productList__container{
    display: flex;
    flex-direction: column;
    gap: 10px;
    margin: 30px auto;
    padding: 7px;
    text-align: center;
}

.productList__container__sort{
    width: 50%;
    text-align: left;
    font-weight: bold;
}

.productList__container__sort i{
  padding-right: 8px;
}

.select-control{
    background: transparent;
    border: none;
}

.productList__spinner-container{
   margin-top: 150px;
}

.dark .select-control{
    background: transparent;
}