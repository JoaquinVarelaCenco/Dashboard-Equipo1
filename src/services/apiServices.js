const url = "http://localhost:5000/api/";
const getProducts = () => {
  return fetch(`${url}product`).then((res) => res.json())
};

const getProductById = (id) => {
  return fetch(`${url}product/${id}`).then((res) => res.json());
};

const deleteProduct = (productId) => {
  return fetch(`${url}product?id=${productId}`, {
    method: "DELETE",
  }).then((res) => res.json());
};

const updateProduct = (body) => {
  return fetch(`${url}product`, {
    method: "PUT",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(body),
  }).then((res) => res.json());
};

const newProduct = (body) => {
  return fetch(`${url}product`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(body),
  }).then((res) => res.json());
};

module.exports = {
  getProducts,
  getProductById,
  deleteProduct,
  updateProduct,
  newProduct,
};
