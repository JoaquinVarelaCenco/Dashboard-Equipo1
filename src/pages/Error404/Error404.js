import { Link } from 'react-router-dom';
import './Error404.css';

const Error404 = () => {
  return (
    <>
        <div className='error404__Main'>
            <div className='error404__top'>
                <p>404</p>
            </div>
            <div className='error404__bottom'>
                <h3>Disculpa, Página No Encontrada</h3>
                <p>La página que deseas acceder no existe</p>
                <Link to="/">
                    <button>Home</button>
                </Link>
            </div>
        </div>
    </>
  )
}

export default Error404