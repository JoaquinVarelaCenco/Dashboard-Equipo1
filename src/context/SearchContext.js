import { useState, createContext, useEffect } from "react";
import { getProducts } from "../services/apiServices";

export const SearchContext = createContext();

export const SearchProvider = ({ children }) => {
  const [products, setProducts] = useState([]);
  const [productsExist, setProductsExist] = useState(true);
  const [helpProducts, setHelpProducts] = useState([]);
  const [searchTermValue, setSearchTermValue] = useState("");
  const [orderBy, setOrderBy] = useState("Mas relevantes");

  const getAllProducts = () => {
    getProducts()
      .then((data) => {
        setProductsExist(true);
        orderProductsPrueba(data);
        setProducts(data);
        setHelpProducts(data);
      })
      .catch(() => {
        setProductsExist(false);
      });
  };

  const filterProducts = (searchTerm) => {
    const newProducts = helpProducts.filter((val) => {
      if (
        searchTerm === "" ||
        val.title.toLowerCase().includes(searchTerm.toLowerCase())
      ) {
        return val;
      } else if (
        val.description.toLowerCase().includes(searchTerm.toLowerCase())
      ) {
        return val;
      }
    });
    return newProducts;
  };

  const handleSearch = (e) => {
    const newFilter = filterProducts(e.target.value);
    setSearchTermValue(e.target.value);
    setProducts(newFilter);
  };

  const orderByHighPrice = (arrayData) => {
    arrayData.sort((a, b) =>
      a.price < b.price ? 1 : a.price > b.price ? -1 : 0
    );
  };

  const orderByLowPrice = (arrayData) => {
    arrayData.sort((a, b) =>
      a.price > b.price ? 1 : a.price < b.price ? -1 : 0
    );
  };

  const orderByMostRelevants = (arrayData) => {
    arrayData.sort((a, b) =>
      a.rating.count < b.rating.count
        ? 1
        : a.rating.count > b.rating.count
        ? -1
        : 0
    );
  };

  const orderByAlphabet = (arrayData) => {
    arrayData.sort((a, b) =>
      a.title > b.title ? 1 : a.title < b.title ? -1 : 0
    );
  };

  const orderProducts = (e) => {
    switch (e.target.value) {
      case "Mayor precio":
        setOrderBy("Mayor precio");
        orderByHighPrice(products);
        break;
      case "Menor precio":
        setOrderBy("Menor precio");
        orderByLowPrice(products);
        break;
      case "Mas relevantes":
        setOrderBy("Mas relevantes");
        orderByMostRelevants(products);
        break;
      case "A-Z":
        setOrderBy("A-Z");
        orderByAlphabet(products);
        break;
    }
  };

  const orderProductsPrueba = (data) => {
    
    switch (orderBy) {
      case "Mayor precio":
        setOrderBy("Mayor precio");
        orderByHighPrice(data);
        break;
      case "Menor precio":
        setOrderBy("Menor precio");
        orderByLowPrice(data);
        break;
      case "Mas relevantes":
        setOrderBy("Mas relevantes");
        orderByMostRelevants(data);
        break;
      case "A-Z":
        setOrderBy("A-Z");
        orderByAlphabet(data);
        break;
    }
  };



  useEffect(() => {
    getAllProducts();
  }, []);

  return (
    <SearchContext.Provider
      value={{
        handleSearch,
        products,
        productsExist,
        getAllProducts,
        helpProducts,
        searchTermValue,
        orderBy,
        orderProducts
      }}
    >
      {children}
    </SearchContext.Provider>
  );
};
