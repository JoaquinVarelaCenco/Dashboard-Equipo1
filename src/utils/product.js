import { deleteProduct } from "../services/apiServices";


export function deleteProductFunction(id) {
    deleteProduct(id).then((res) => {
    });
}

export const sortByCount = (products) =>{
    const sorted = products.sort((a,b) =>{
    if(a.rating.count < b.rating.count){
        return 1;
    }else  if(a.rating.count > b.rating.count){
        return -1;
    }else{
        return 0;
    }
    });
    return sorted;
}

export const sortByStock = (products) =>{
    const sorted = products.sort((a,b) =>{
    if(a.stock < b.stock){
        return 1;
    }else  if(a.stock > b.stock){
        return -1;
    }else{
        return 0;
    }
    });
    return sorted;
}