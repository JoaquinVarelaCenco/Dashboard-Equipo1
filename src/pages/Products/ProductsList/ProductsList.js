import { useContext, useEffect, useState } from "react";
import { SearchContext } from "../../../context/SearchContext";
import ProductCard from "../../../components/ProductCard/ProductCard";
import Spinner from "../../../components/Spinner/Spinner";
import WarningMessage from "../../../components/WarningMessage/WarningMessage";
import ProductsSortMenu from "../../../components/ProductsSortMenu/ProductsSortMenu";
import "./ProductsList.css";

const ProductsList = () => {
  const {
        products,
        productsExist,
        getAllProducts,
        searchTermValue,
  } = useContext(SearchContext);

  useEffect(() => {
    getAllProducts();
  }, []);

  return (
    <div className="productList__container">
      <ProductsSortMenu/>
      {productsExist ? (
        products.length !== 0 ? (
          products.map((p) => (
            
            <ProductCard
              title={p.title}
              price={p.price}
              key={p.id}
              image={p.images!=null? p.images[0] : null}
              id={p.id}
            />
          ))
        ) : searchTermValue !== "" && products.length === 0 ? (
          <WarningMessage
            text="No hay productos que coincidan con tu busqueda"
            search={true}
          />
        ) : (
          <div className="productList__spinner-container">
            <Spinner />
          </div>
        )
      ) : (
        <WarningMessage text="No existen productos" search={false} />
      )}
    </div>
  );
};

export default ProductsList;
