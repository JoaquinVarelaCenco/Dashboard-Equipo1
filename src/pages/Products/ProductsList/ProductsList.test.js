import { render, act, screen, cleanup } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { MemoryRouter } from "react-router-dom";
import { SearchProvider } from "../../../context/SearchContext";
import productsData from "../../../mockData/productsData";
import { getProducts } from "../../../services/apiServices";
import { orderByAlphabet, orderByHighPrice, orderByLowPrice, orderByMostRelevants } from "../../../utils/product";
import Error404 from "../../Error404/Error404";
import ProductsList from "./ProductsList";

jest.mock("../../../services/apiServices");

describe("Test formulario de productos", () => {
  let component;

  beforeEach(async () => {

    getProducts.mockResolvedValue([...productsData])

    await act(async () => {
      component = await render(
      <MemoryRouter><ProductsList /></MemoryRouter>, { wrapper: SearchProvider });

    });
  });

  it("Se debe renderizar correctamente",  () => {
    const { container } = component;
    expect(container).toMatchSnapshot();
  });

  it("Se debe renderizar correctamente los items",  () => {
      const items = screen.getAllByRole('article') 
      expect(items.length).toBe(productsData.length)
  });

  it("Seleccion de filtrado predeterminada",  () => {
     const optionMasRelevantes = screen.getByRole('option', { name: 'Mas relevantes' }).selected;
     const optionMayorPrecio = screen.getByRole('option', { name: 'Mayor precio' }).selected;
     const optionMenorPrecio = screen.getByRole('option', { name: 'Menor precio' }).selected;
     const optionAZ = screen.getByRole('option', { name: 'A-Z' }).selected;

     expect(optionMasRelevantes).toBe(true)
     expect(optionMayorPrecio).toBe(false)
     expect(optionMenorPrecio).toBe(false)
     expect(optionAZ).toBe(false)
});

it("Seleccion de filtrado por interaccion del usuario por Mayor precio",  () => {

  
  userEvent.selectOptions(
    screen.getByRole('combobox'),
    screen.getByRole('option', { name: 'Mayor precio' }),
  )

  const optionMasRelevantes = screen.getByRole('option', { name: 'Mas relevantes' }).selected;
  const optionMayorPrecio = screen.getByRole('option', { name: 'Mayor precio' }).selected;
  const optionMenorPrecio = screen.getByRole('option', { name: 'Menor precio' }).selected;
  const optionAZ = screen.getByRole('option', { name: 'A-Z' }).selected;

  expect(optionMayorPrecio).toBe(true)
  expect(optionMenorPrecio).toBe(false)
  expect(optionAZ).toBe(false)
  expect(optionMasRelevantes).toBe(false)
});

it("Se deben renderizar los items filtrados por Mayor precio",  () => {

  userEvent.selectOptions(
    screen.getByRole('combobox'),
    screen.getByRole('option', { name: 'Mayor precio' }),
  );

  const optionMayorPrecio = screen.getByRole('option', { name: 'Mayor precio' }).selected;

  expect(optionMayorPrecio).toBe(true);

  const arrayProd = screen.getAllByTestId('id-product');
  const arrayProdPivot = arrayProd.map(a => a.innerHTML);
  let orderProductsData = orderByHighPrice(productsData);
  orderProductsData = orderProductsData.map(b => `#${b.id}`);
  expect(orderProductsData).toEqual(arrayProdPivot);
});

it("Seleccion de filtrado por interaccion del usuario por Menor precio",  () => {

  userEvent.selectOptions(
    screen.getByRole('combobox'),
    screen.getByRole('option', { name: 'Menor precio' }),
  )

  const optionMasRelevantes = screen.getByRole('option', { name: 'Mas relevantes' }).selected;
  const optionMayorPrecio = screen.getByRole('option', { name: 'Mayor precio' }).selected;
  const optionMenorPrecio = screen.getByRole('option', { name: 'Menor precio' }).selected;
  const optionAZ = screen.getByRole('option', { name: 'A-Z' }).selected;

  expect(optionMayorPrecio).toBe(false)
  expect(optionMenorPrecio).toBe(true)
  expect(optionAZ).toBe(false)
  expect(optionMasRelevantes).toBe(false)

});

it("Se deben renderizar los items filtrados por Menor precio",  () => {

  userEvent.selectOptions(
    screen.getByRole('combobox'),
    screen.getByRole('option', { name: 'Menor precio' }),
  );

  const optionMayorPrecio = screen.getByRole('option', { name: 'Menor precio' }).selected;

  expect(optionMayorPrecio).toBe(true);

  const arrayProd = screen.getAllByTestId('id-product');
  const arrayProdPivot = arrayProd.map(a => a.innerHTML);
  let orderProductsData = orderByLowPrice(productsData);
  orderProductsData = orderProductsData.map(b => `#${b.id}`);
  expect(orderProductsData).toEqual(arrayProdPivot);
});

it("Seleccion de filtrado por interaccion del usuario por Mas relevantes",  () => {

  userEvent.selectOptions(
    screen.getByRole('combobox'),
    screen.getByRole('option', { name: 'Mas relevantes' }),
  )

  const optionMasRelevantes = screen.getByRole('option', { name: 'Mas relevantes' }).selected;
  const optionMayorPrecio = screen.getByRole('option', { name: 'Mayor precio' }).selected;
  const optionMenorPrecio = screen.getByRole('option', { name: 'Menor precio' }).selected;
  const optionAZ = screen.getByRole('option', { name: 'A-Z' }).selected;

  expect(optionMayorPrecio).toBe(false)
  expect(optionMenorPrecio).toBe(false)
  expect(optionAZ).toBe(false)
  expect(optionMasRelevantes).toBe(true)

});

it("Se deben renderizar los items filtrados por Mas relevantes",  () => {

  userEvent.selectOptions(
    screen.getByRole('combobox'),
    screen.getByRole('option', { name: 'Mas relevantes' }),
  );

  const optionMayorPrecio = screen.getByRole('option', { name: 'Mas relevantes' }).selected;

  expect(optionMayorPrecio).toBe(true);

  const arrayProd = screen.getAllByTestId('id-product');
  const arrayProdPivot = arrayProd.map(a => a.innerHTML);
  let orderProductsData = orderByMostRelevants(productsData);
  orderProductsData = orderProductsData.map(b => `#${b.id}`);
  expect(orderProductsData).toEqual(arrayProdPivot);
});

it("Seleccion de filtrado por interaccion del usuario por A-Z",  () => {

  userEvent.selectOptions(
    screen.getByRole('combobox'),
    screen.getByRole('option', { name: 'A-Z' }),
  )

  const optionMasRelevantes = screen.getByRole('option', { name: 'Mas relevantes' }).selected;
  const optionMayorPrecio = screen.getByRole('option', { name: 'Mayor precio' }).selected;
  const optionMenorPrecio = screen.getByRole('option', { name: 'Menor precio' }).selected;
  const optionAZ = screen.getByRole('option', { name: 'A-Z' }).selected;

  expect(optionMayorPrecio).toBe(false)
  expect(optionMenorPrecio).toBe(false)
  expect(optionAZ).toBe(true)
  expect(optionMasRelevantes).toBe(false)

});

it("Se deben renderizar los items filtrados por orden alfabetico de A a la Z",  () => {

  userEvent.selectOptions(
    screen.getByRole('combobox'),
    screen.getByRole('option', { name: 'A-Z' }),
  );

  const optionMayorPrecio = screen.getByRole('option', { name: 'A-Z' }).selected;

  expect(optionMayorPrecio).toBe(true);

  const arrayProd = screen.getAllByTestId('id-product');
  const arrayProdPivot = arrayProd.map(a => a.innerHTML);
  let orderProductsData = orderByAlphabet(productsData);
  orderProductsData = orderProductsData.map(b => `#${b.id}`);
  expect(orderProductsData).toEqual(arrayProdPivot);
});
// test.only('Se debe renderizar el warning message cuando no hay nada en la lista de productos', async () => { 
//   cleanup()
//   await getProducts.mockRejectedValue([]);
//    render(
//     <MemoryRouter><ProductsList/></MemoryRouter>, { wrapper: SearchProvider });
// screen.debug()
//   });
 })


