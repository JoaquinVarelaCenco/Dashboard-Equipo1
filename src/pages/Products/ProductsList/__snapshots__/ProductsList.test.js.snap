// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test formulario de productos Se debe renderizar correctamente 1`] = `
<div>
  <div
    class="productList__container"
  >
    <div
      class="productList__container__sort"
    >
      <div>
        <i>
          Ordenar por
        </i>
        <select
          class="select-control-products"
          name="category"
        >
          <option
            value="Mayor precio"
          >
            Mayor precio
          </option>
          <option
            value="Menor precio"
          >
            Menor precio
          </option>
          <option
            value="Mas relevantes"
          >
            Mas relevantes
          </option>
          <option
            value="A-Z"
          >
            A-Z
          </option>
        </select>
      </div>
      <i>
        5
         resultados
      </i>
    </div>
    <a
      class="product-card-link"
      href="/products/24"
    >
      <article
        class="main-product__related-product-card"
      >
        <div
          class="main-product__imageAndDesc"
        >
          <div
            class="section-article__image"
          >
            <img
              alt="Imagen de producto"
              class="imagenPrincipal hideImage"
              src="https://dummyjson.com/image/i/products/24/1.jpg"
            />
            <div
              class="spinnerWrapper showSpinner"
            >
              <div
                class="spinnerContainer"
              >
                <div
                  class="lds-ripple"
                >
                  <div />
                  <div />
                </div>
              </div>
            </div>
          </div>
          <div
            class="section-article__desc"
          >
            <div
              class="desc"
            >
              <p>
                cereals muesli fruit nuts
              </p>
            </div>
            <div
              class="section-article__price"
            >
              <p
                class="valorNumerico"
              >
                #
                24
              </p>
            </div>
          </div>
        </div>
        <div
          class="section-article__goToProduct"
        >
          <img
            alt=""
            src="chevron-right-test.svg"
          />
        </div>
      </article>
    </a>
    <a
      class="product-card-link"
      href="/products/23"
    >
      <article
        class="main-product__related-product-card"
      >
        <div
          class="main-product__imageAndDesc"
        >
          <div
            class="section-article__image"
          >
            <img
              alt="Imagen de producto"
              class="imagenPrincipal hideImage"
              src="https://dummyjson.com/image/i/products/23/1.jpg"
            />
            <div
              class="spinnerWrapper showSpinner"
            >
              <div
                class="spinnerContainer"
              >
                <div
                  class="lds-ripple"
                >
                  <div />
                  <div />
                </div>
              </div>
            </div>
          </div>
          <div
            class="section-article__desc"
          >
            <div
              class="desc"
            >
              <p>
                Orange Essence Food Flavou
              </p>
            </div>
            <div
              class="section-article__price"
            >
              <p
                class="valorNumerico"
              >
                #
                23
              </p>
            </div>
          </div>
        </div>
        <div
          class="section-article__goToProduct"
        >
          <img
            alt=""
            src="chevron-right-test.svg"
          />
        </div>
      </article>
    </a>
    <a
      class="product-card-link"
      href="/products/3"
    >
      <article
        class="main-product__related-product-card"
      >
        <div
          class="main-product__imageAndDesc"
        >
          <div
            class="section-article__image"
          >
            <img
              alt="Imagen de producto"
              class="imagenPrincipal hideImage"
              src="no-image.jpeg"
            />
            <div
              class="spinnerWrapper showSpinner"
            >
              <div
                class="spinnerContainer"
              >
                <div
                  class="lds-ripple"
                >
                  <div />
                  <div />
                </div>
              </div>
            </div>
          </div>
          <div
            class="section-article__desc"
          >
            <div
              class="desc"
            >
              <p>
                Samsung Universe 9
              </p>
            </div>
            <div
              class="section-article__price"
            >
              <p
                class="valorNumerico"
              >
                #
                3
              </p>
            </div>
          </div>
        </div>
        <div
          class="section-article__goToProduct"
        >
          <img
            alt=""
            src="chevron-right-test.svg"
          />
        </div>
      </article>
    </a>
    <a
      class="product-card-link"
      href="/products/25"
    >
      <article
        class="main-product__related-product-card"
      >
        <div
          class="main-product__imageAndDesc"
        >
          <div
            class="section-article__image"
          >
            <img
              alt="Imagen de producto"
              class="imagenPrincipal hideImage"
              src="https://dummyjson.com/image/i/products/25/1.png"
            />
            <div
              class="spinnerWrapper showSpinner"
            >
              <div
                class="spinnerContainer"
              >
                <div
                  class="lds-ripple"
                >
                  <div />
                  <div />
                </div>
              </div>
            </div>
          </div>
          <div
            class="section-article__desc"
          >
            <div
              class="desc"
            >
              <p>
                Gulab Powder 50 Gram
              </p>
            </div>
            <div
              class="section-article__price"
            >
              <p
                class="valorNumerico"
              >
                #
                25
              </p>
            </div>
          </div>
        </div>
        <div
          class="section-article__goToProduct"
        >
          <img
            alt=""
            src="chevron-right-test.svg"
          />
        </div>
      </article>
    </a>
    <a
      class="product-card-link"
      href="/products/2"
    >
      <article
        class="main-product__related-product-card"
      >
        <div
          class="main-product__imageAndDesc"
        >
          <div
            class="section-article__image"
          >
            <img
              alt="Imagen de producto"
              class="imagenPrincipal hideImage"
              src="no-image.jpeg"
            />
            <div
              class="spinnerWrapper showSpinner"
            >
              <div
                class="spinnerContainer"
              >
                <div
                  class="lds-ripple"
                >
                  <div />
                  <div />
                </div>
              </div>
            </div>
          </div>
          <div
            class="section-article__desc"
          >
            <div
              class="desc"
            >
              <p>
                iPhone X
              </p>
            </div>
            <div
              class="section-article__price"
            >
              <p
                class="valorNumerico"
              >
                #
                2
              </p>
            </div>
          </div>
        </div>
        <div
          class="section-article__goToProduct"
        >
          <img
            alt=""
            src="chevron-right-test.svg"
          />
        </div>
      </article>
    </a>
  </div>
</div>
`;
